src/Components/NoteItem/NoteItem.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react'
+import {BsTrash3} from 'react-icons/bs'
+import './noteItem.css'
+import DeleteModel from '../DeleteModel/DeleteModel'
+import UpdateModel from '../UpdateModel/UpdateModel'
+function NoteItem({note , onDelete , onUpdate}) {
+  const [showdeleteModel, setShowdeleteModel] = useState(false)
+  const [showUpdateModel, setShowUpdateModel] = useState(false)
```

Comment:
I prefer the first suggestion

```suggestion
  const [deleteModelOpened, setDeleteModelOpened] = useState(false)
  const [updateModelOpened, setUpdateModelOpened] = useState(false)
```
OR
```suggestion
  const [modelOpened, setModelOpened] = useState<string>(null); // it can take the opened modal name, and below we can use this state to check if it's opened.
  
 {modelOpened === "DELETE" && <DeleteModel />}
 {modelOpened === "UPDATE" && <UpdateModel/>}
```